### PR TITLE
FMv1 to FMv2 migrator

### PR DESCRIPTION
**Pull request by connorwstein**
The approach taken is to read the v1 spec from the app, manipulate it, pipe it back through the app code to create a v2 spec, then archive the v1 job. This way we ensure all the v2 data input validation is satisfied. At some point once all v1 jobs are migrated we can just drop the v1 tables to remove any archived jobs. Some notes about FM v1 jobs specifically (gathered from integrations):
- The fromAddress is not currently used in EthTx tasks, so we do not need to immediately add support for this in FMv2
- The per FM job confirmations in EthTx tasks are also currently unused, so we do not need to immediately add support for this in FMv2
- All FM specs use bridges which return a number specifically (no json parsing required) and then have 3 tasks: Multiply, EthUint256 (must be this because thats the contract arg type) and EthTx
- We are ok with not migrating the v1 job spec errors and just starting to produce v2 spec errors after the job has been migrated. 

TODO
- Acceptance testing on some live testnet v1 FM jobs

New CLI command is present to help you move jobs over is below. 
image

### CODE REVIEW

File: core/services/pipeline/graph.go [hunk 1]
```diff
@@ -123,55 +124,78 @@ func (g TaskDAG) outputs() []*taskDAGNode {
...
 }
 
-func (n *taskDAGNode) inputs() []*taskDAGNode {
-	var nodes []*taskDAGNode
+func (n *TaskDAGNode) Attributes() []encoding.Attribute {
+	var r []encoding.Attribute
+	for k, v := range n.attrs {
+		r = append(r, encoding.Attribute{Key: k, Value: v})
+	}
+	sort.Slice(r, func(i, j int) bool {
```
Comment (connorwstein) [inline core/services/pipeline/graph.go:171]: sorting just to ensure deterministic ordering of node attributes, helps with the expected test dags

File: core/store/presenters/presenters.go [hunk 2]
```diff
@@ -334,9 +336,11 @@ func (i Initiator) MarshalJSON() ([]byte, error) {
 	}
 
 	return json.Marshal(&struct {
+		ID     int64       `json:"id"`
```
Comment (connorwstein) [inline core/store/presenters/presenters.go:339]: It wasn't clear to me why we left this out before - seems confusing to return an empty ID/jobID in some parts of the JSON response?

File: core/services/pipeline/graph.go (hunk 1 above)
Comment (samsondav) [reply core/services/pipeline/graph.go:171]: Makes sense, let's add this as a comment tho

File: core/web/migrate_controller.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+)
+
+type MigrateController struct {
+	App chainlink.Application
+}
+
+// Creates a v2 job from a v1 job ID.
+// Example:
+//  "POST <application>/migrate/123e4567-e89b-12d3-a456-426614174000"
+// Where "123e4567-e89b-12d3-a456-426614174000" is a v1 job ID.
```
Comment (samsondav) [inline core/web/migrate_controller.go:35]: Presumably this is irreversible? Is there an easy way for a node operator to "undo" a migration?

**Review by samsondav — DISMISSED** [review]
Nice work, from my understanding we still don't have any link between eth_tx and direct request v2 or fm v2 do we?

**Comment by samsondav** [comment]
Ah, this will need a changelog entry also

File: core/web/migrate_controller.go (hunk 3 above)
Comment (connorwstein) [reply core/web/migrate_controller.go:35]: No easy way - it's roll forward

Comment (samsondav) [reply core/web/migrate_controller.go:35]: OK as long as we explain this in the Changelog

**Review by samsondav — DISMISSED** [review]

File: docs/CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add `MockOracle.sol` for testing contracts
+- New CLI command to convert v1 flux monitor jobs (JSON) to 
+v2 flux monitor jobs (TOML). Running it will archive the v1 
+job and create a new v2 job. Example:
+```
+chainlink jobs migrate 123e4567-e89b-12d3-a456-426614174000
+```
```
Comment (samsondav) [inline docs/CHANGELOG.md:18]: Better add here "to undo the migration, XXX"

**Review by samsondav — APPROVED** [review]

**Pull request merged**